subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DirectorySnapshotterTest.groovy

```diff
@@ -155,6 +155,26 @@ class DirectorySnapshotterTest extends Specification {
         ] as Set
     }
 
+    @Requires(TestPrecondition.FILE_PERMISSIONS)
+    def "broken symlinks are snapshotted as missing"() {
+        def rootDir = tmpDir.createDir("root")
+        rootDir.file('brokenSymlink').createLink("linkTarget")
+        assert rootDir.listFiles()*.exists() == [false]
+        def predicate = directoryWalkerPredicate(new PatternSet())
```

Comment:
Why do you use a non-null predicate here?

Reply:
Leftover, but it wasn't working with `null` for some reason ... anyway, it seems to be now, fixed :)

Reply:
Done